File: task-03/Ruby/subtask-04.rb
def print_star_pattern(n)
  pattern = ""

  (1..n).step(2) do |i|
    pattern += ' ' * ((n - i) / 2) + '*' * i + "\n"
  end

  (n-2).step(1, -2) do |i|
    pattern += ' ' * ((n - i) / 2) + '*' * i + "\n"
  end

  pattern
end

s = File.read("input2.txt").to_i

pattern = print_star_pattern(s)

File.open("output2.txt", "w") do |f|
  f.write(pattern)
end

File: task-03/Python/subtask-04.py
def print_star_pattern(n):

    pattern = ""

    for i in range(1, n + 1, 2):
        pattern += ' ' * ((n - i) // 2) + '*' * i + '\n'


    for i in range(n-2, 0, -2):
        pattern += ' ' * ((n - i) // 2) + '*' * i + '\n'

    return pattern

f = open("input2.txt",'r')
s = int(f.read())
f.close()

pattern = print_star_pattern(s)

f = open("output2.txt",'w')
f.write(pattern)
f.close()
